bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyChannelDefinitionsDTO.java

```diff
@@ -0,0 +1,326 @@
+                CHANNEL_SENSOR_MOTION);
+        addChannel(thing, newChannels, sdata.charger != null, CHANNEL_GROUP_DEV_STATUS, CHANNEL_DEVST_CHARGER);
+        addChannel(thing, newChannels, sdata.sensorError != null, CHANNEL_GROUP_SENSOR, CHANNEL_SENSOR_ERROR);
+        addChannel(thing, newChannels, true, CHANNEL_GROUP_SENSOR, CHANNEL_LAST_UPDATE);
+        addChannel(thing, newChannels, sdata.actReasons != null, CHANNEL_GROUP_DEV_STATUS, CHANNEL_DEVST_WAKEUP);
+
+        // Battery
+        if (sdata.bat != null) {
+            addChannel(thing, newChannels, sdata.bat.value != null, CHANNEL_GROUP_BATTERY, CHANNEL_SENSOR_BAT_LEVEL);
+            addChannel(thing, newChannels, sdata.bat.value != null, CHANNEL_GROUP_BATTERY, CHANNEL_SENSOR_BAT_LOW);
+            addChannel(thing, newChannels, sdata.bat.voltage != null, CHANNEL_GROUP_BATTERY, CHANNEL_SENSOR_BAT_VOLT);
+        }
+        return newChannels;
+    }
+
+    private static void addChannel(Thing thing, Map<String, Channel> newChannels, boolean supported, String group,
+            String channelName) throws IllegalArgumentException {
+        if (supported) {
+            final String channelId = group + "#" + channelName;
+            final ShellyChannel channelDef = getDefinition(channelId);
+            final ChannelUID channelUID = new ChannelUID(thing.getUID(), channelId);
+            final ChannelTypeUID channelTypeUID = channelDef.typeId.contains("system:")
+                    ? new ChannelTypeUID(channelDef.typeId)
+                    : new ChannelTypeUID(BINDING_ID, channelDef.typeId);
+
+            // Channel channel = ChannelBuilder.create(channelUID, channelId).withType(channelTypeUID)
+            // .withLabel(channelDef.label).withDescription(channelDef.description).build();
+            Channel channel = ChannelBuilder.create(channelUID, channelDef.itemType).withType(channelTypeUID).build();
+            newChannels.put(channelId, channel);
+        }
+    }
+
+    public static String ITEM_TYPE_NUMBER = "Number";
+    public static String ITEM_TYPE_STRING = "String";
+    public static String ITEM_TYPE_DATETIME = "DateTime";
+    public static String ITEM_TYPE_TEMP = "Number:Temperature";
+    public static String ITEM_TYPE_LUX = "Number:Illuminance";
+    public static String ITEM_TYPE_POWER = "Number:Power";
+    public static String ITEM_TYPE_ENERGY = "Number:Energy";
+    public static String ITEM_TYPE_VOLT = "Number:ElectricPotential";
+    public static String ITEM_TYPE_AMP = "Number:ElectricPotential";
+    public static String ITEM_TYPE_PERCENT = "Number:Dimensionless";
+    public static String ITEM_TYPE_SWITCH = "Switch";
+    public static String ITEM_TYPE_CONTACT = "Contact";
+
+    public static String PREFIX_GROUP = "definitions.shelly.group.";
+    public static String PREFIX_CHANNEL = "channel-type.shelly.";
+    public static String SUFFIX_LABEL = ".label";
+    public static String SUFFIX_DESCR = ".description";
```

Comment:
All of these fields should be final and moved to the top of this class.

Reply:
done